src/individual.c

```diff
@@ -246,6 +257,58 @@ void set_case( individual *indiv, int time )
 	indiv->infection_events->times[CASE] = time;
 }
 
+/*****************************************************************************************
+*  Name:		set_waiting
+*  Description: sets a person to be added to the hospital waiting list
+*  Returns:		void
+******************************************************************************************/
+void set_waiting( individual *indiv, parameters* params, int time )
+{
+	indiv->hospital_state = WAITING;
+}
+
+/*****************************************************************************************
+*  Name:		set_general_admission
+*  Description: sets a person to be added to a general ward
+*  Returns:		void
+******************************************************************************************/
+void set_general_admission( individual *indiv, parameters* params, int time )
+{
+	indiv->hospital_state = GENERAL;
+}
+
+/*****************************************************************************************
+*  Name:		set_icu_admission
+*  Description: sets a person to be added to an ICU
+*  Returns:		void
+******************************************************************************************/
+void set_icu_admission( individual *indiv, parameters* params, int time )
+{
+	indiv->hospital_state = ICU;
+}
+
+/*****************************************************************************************
+*  Name:		set_mortuary_admission
+*  Description: sets a dead person to be added to the mortuary
+*  Returns:		void
+******************************************************************************************/
+void set_mortuary_admission( individual *indiv, parameters* params, int time )
+{
+	indiv->hospital_state = MORTUARY;
+	indiv->current_hospital_event = NULL;
+}
+
+/*****************************************************************************************
+*  Name:		set_discharged
+*  Description: sets a recovered person to be discharged from the hospital.
+*  Returns:		void
+******************************************************************************************/
+void set_discharged( individual *indiv, parameters* params, int time )
+{
+	indiv->hospital_state = DISCHARGED;
```

Comment:
Could have people transition back to NOT_IN_HOSPITAL. New structure called infection events, which records times and states of a single infection, if people become susceptible, then this opens up the possibility for them to be reinfected.

Alternatively, have them just get reset to NOT_IN_HOSPITAL